backend/src/estark/mod.rs

```diff
@@ -0,0 +1,223 @@
+mod json_exporter;
+#[cfg(feature = "estark-polygon")]
+pub mod polygon_wrapper;
+pub mod starky_wrapper;
+
+use std::{
+    fs::File,
+    io::{self, BufWriter, Write},
+    iter::{once, repeat},
+    path::Path,
+};
+
+use crate::{Backend, BackendFactory, Error, Proof};
+use powdr_ast::analyzed::Analyzed;
+
+use powdr_executor::witgen::WitgenCallback;
+use powdr_number::{write_polys_file, DegreeType, FieldElement};
+use serde::Serialize;
+use starky::types::{StarkStruct, Step, PIL};
+
+fn create_stark_struct(degree: DegreeType) -> StarkStruct {
+    assert!(degree > 1);
+    let n_bits = (DegreeType::BITS - (degree - 1).leading_zeros()) as usize;
+    let n_bits_ext = n_bits + 1;
+
+    let steps = (2..=n_bits_ext)
+        .rev()
+        .step_by(4)
+        .map(|b| Step { nBits: b })
+        .collect();
+
+    StarkStruct {
+        nBits: n_bits,
+        nBitsExt: n_bits_ext,
+        nQueries: 2,
+        verificationHashType: "GL".to_owned(),
+        steps,
+    }
+}
+
+fn pil_hack_fix<'a, F: FieldElement>(
```

Comment:
```suggestion
fn pil_first_step_hack<'a, F: FieldElement>(
```
or something a bit more descriptive?